File: frontend/src/types/event_log.ts
export type EventLogCategory =
  | "reservation_skipped"
  | "assembly_stock_shortfall"
  | "stock_operation_failed"
  | "retention_cleanup"
  | "auth_error";
export type EventLogLevel = "info" | "warning" | "error";

export interface EventLog {
  id: number;
  category: EventLogCategory;
  level: EventLogLevel;
  message: string;
  order_id: number | null;
  order_unique_key: string | null;
  item_id: string | null;
  shop_id: number | null;
  created_at: string;
  // 同一(category, order_id, item_id)が繰り返し発生した回数。集約対象外のカテゴリは常に1
  occurrence_count: number;
  last_occurred_at: string;
}

export interface EventLogListResult {
  items: EventLog[];
  total: number;
  page: number;
}

File: backend/app/schemas/event_log.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class EventLogRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    category: str
    level: str
    message: str
    order_id: int | None
    order_unique_key: str | None = None
    item_id: str | None
    shop_id: int | None
    created_at: datetime
    occurrence_count: int
    last_occurred_at: datetime


class EventLogListRead(BaseModel):
    items: list[EventLogRead]
    total: int
    # highlight指定時、そのイベントログが含まれるページ番号(1始まり)。フロントエンドが
    # 自分でページを計算しなくても該当ページへ自動的に合わせられるようにする
    page: int = 1
